pkg/repositories/repositories.go

```diff
@@ -105,13 +106,16 @@ func (r *Repositories) withLock(fn func() error) error {
 		return err
 	}
 
+	blocker := func() error {
+		time.Sleep(time.Millisecond)
+		return nil
+	}
```

Comment:
This was pretty much stolen verbatim from the unit tests in `gofslock`, but they seemed straightforward enough.

Reply:
lgtm!